### PR TITLE
feat: show warning whrn GRPC method is disallowed

### PR DESCRIPTION
**Pull request by stringhandler**
Description
---
Adds a log and console warning if a GRPC method is called but not allowed

### CODE REVIEW

**Review by hansieodendaal — APPROVED** [review]
utACK

**Review by brianp — APPROVED** [review]
I of course debated internally if `info` makes more sense then `warn` but I suppose someone is actively calling a method that has been disabled then a warn can be warranted.

File: applications/minotari_node/src/grpc/base_node_grpc_server.rs [hunk 1]
```diff
@@ -151,6 +151,17 @@ impl BaseNodeGrpcServer {
...
+    fn check_method_enabled<T>(&self, method: GrpcMethod) -> Option<Result<Response<T>, Status>> {
+        if !self.is_method_enabled(method) {
+            warn!(target: LOG_TARGET, "`{}` method called but it is not allowed. Allow it in the config file or start the node with a different set of CLI options", method);
+            return Some(Err(Status::permission_denied(format!(
+                "`{}` method not made available",
+                method
+            ))));
+        }
+        None
+    }
```
Comment (delta1) [inline applications/minotari_node/src/grpc/base_node_grpc_server.rs:164]: why not return a `Result` so you can just do 

```
self.check_method_enabled(..)?
```

Comment (stringhandler) [reply applications/minotari_node/src/grpc/base_node_grpc_server.rs:164]: it has to return the Status to the client.

Comment (delta1) [reply applications/minotari_node/src/grpc/base_node_grpc_server.rs:164]: oh right

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**